src/components/LoginForm/LoginForm.tsx

```diff
@@ -11,9 +11,9 @@ import './LoginForm.css';
 
 export default function LoginForm(): JSX.Element {
     const history = useHistory();
-    const user = useUserContext();
+    const { pending, user } = useUserContext();
     React.useEffect(() => {
-        if (user) {
+        if (!pending && user) {
```

Comment:
What effect does `pending` have on this component?

Reply:
What I was thinking is that when we go from `{pending: false, user: null}` -> `{pending:true, user: null}` after the user enters their email and password to log in rather that just flashing to a logged in state we can show a spinner here when user credentials are being fetched.